spec/unit/autodiscovery.spec.js

```diff
@@ -94,7 +94,7 @@ describe("AutoDiscovery", function() {
                 const expected = {
                     "m.homeserver": {
                         state: "FAIL_PROMPT",
-                        error: "Invalid homeserver discovery response",
+                        error: "Failed to get autodiscovery configuration from server",
```

Comment:
It seems like maybe these strings should be defined as constants in the `AutoDiscovery` class, so that the unit tests can use those rather than trying to keep all the strings in sync.  Or else the tests shouldn't care about the human-readable strings.  Maybe not worth blocking this PR, though.

Reply:
The PR lgtm otherwise